tsk/fs/exfatfs_meta.c

```diff
@@ -50,12 +50,10 @@ exfatfs_is_cluster_alloc(FATFS_INFO *a_fatfs, TSK_DADDR_T a_cluster_addr)
     uint8_t bitmap_byte;
     ssize_t bytes_read = 0;
 
-    assert(a_fatfs != NULL);
     if (fatfs_ptr_arg_is_null(a_fatfs, "a_fatfs", func_name)) {
         return -1;
     }
 
-    assert((a_cluster_addr >= FATFS_FIRST_CLUSTER_ADDR) && (a_cluster_addr <= a_fatfs->lastclust));
     if ((a_cluster_addr < FATFS_FIRST_CLUSTER_ADDR) || (a_cluster_addr > a_fatfs->lastclust)) {
```

Comment:
These conditions are slightly different - is there a reason that the assertion here catches if `a_cluster_addr == FATFS_FIRST_CLUSTER_ADDR == a_fatfs->lastclust`? I think this might guard against an OOB read if the cluster address is the address of the last cluster.

Reply:
> These conditions are slightly different 

Yes but aren't they each-others inverse logic ? Lower bound `FATFS_FIRST_CLUSTER_ADDR` upper bound `a_fatfs->lastclust`. The assert will raise if the condition is false, the if will enter the block if the condition is true.

Reply:
You're right - brain fart, ignore me!

LGTM